src/Data/Graph.purs

```diff
@@ -1,152 +1,107 @@
 -- | A data structure and functions for graphs
 
 module Data.Graph
-  ( Edge(..)
-  , Graph(..)
-  , SCC(..)
+  ( Graph
```

Comment:
Do we need to hide the graph constructor? Unless using a newtype to enforce an invariant via smart constructor I prefer to always export constructors as there seems no reason not to (and potentially allows for unforseen use cases).